src/merkle/smt/full/mod.rs

```diff
@@ -101,6 +101,36 @@ impl Smt {
         Ok(tree)
     }
 
+    /// The parallel version of [`Smt::with_entries()`].
+    ///
+    /// Returns a new [`Smt`] instantiated with leaves set as specified by the provided entries,
+    /// constructed in parallel.
+    ///
+    /// All leaves omitted from the entries list are set to [Self::EMPTY_VALUE].
+    #[cfg(feature = "concurrent")]
+    pub fn with_entries_par(
+        entries: impl IntoIterator<Item = (RpoDigest, Word)>,
+    ) -> Result<Self, MerkleError> {
+        <Self as SparseMerkleTree<SMT_DEPTH>>::with_entries_par(Vec::from_iter(entries))
+    }
```

Comment:
I would not make this into a separate function. Instead, I would make it so that when `concurrent` feature is on, `with_entries()` uses this implementation. This way, the user can always use `with_entries()` function and control how it works via the `concurrent` feature.